client/app/components/Modal/index.jsx

```diff
@@ -97,23 +98,53 @@ export class Modal extends React.Component<Props, State> {
     this.setState({ open: !open });
   };
 
-  render() {
+  resolveElement = () => {
     const { element, elementId } = this.props;
+    let Component = null;
```

Comment:
You don't have to initialize these as null. I think you can just do `let Component;` and `let renderComponent;`.